cmd/crictl/exec.go

```diff
@@ -178,37 +196,74 @@ var runtimeExecCommand = &cli.Command{
 			transport: c.String(transportFlag),
 		}
 
-		for _, id := range ids {
-			opts.id = id
+		maxParallel := 1
+		if c.Bool("parallel") {
+			maxParallel = runtime.NumCPU()
+		}
+
+		results := mapParallel(ids, maxParallel, func(id string) error {
+			optsCopy := *&opts
+			optsCopy.id = id
 
 			if outputContainerID {
 				fmt.Println(id + ":")
 			}
-
 			if c.Bool("sync") {
-				exitCode, err := ExecSync(runtimeClient, opts)
+				exitCode, err := ExecSync(runtimeClient, optsCopy)
 				if err != nil {
-					return fmt.Errorf("execing command in container synchronously: %w", err)
+					return fmt.Errorf("execing command in container %s synchronously: %w", id, err)
 				}
 				if exitCode != 0 {
 					return cli.Exit("non-zero exit code", exitCode)
 				}
-				continue
+			} else {
+				ctx, cancel := context.WithCancel(c.Context)
+				defer cancel()
+				err = Exec(ctx, runtimeClient, optsCopy)
+				if err != nil {
+					return fmt.Errorf("execing command in container %s: %w", id, err)
+				}
 			}
 
-			ctx, cancel := context.WithCancel(c.Context)
-			defer cancel()
-			err = Exec(ctx, runtimeClient, opts)
-			if err != nil {
-				return fmt.Errorf("execing command in container: %w", err)
-			}
+			return nil
+		})
+
+		errs := errors.Join(results...)
 
+		if c.Bool("ignore-errors") {
+			if !quiet {
+				logrus.Warnf("Ignoring errors: %v", errs)
+			}
```

Comment:
This can be moved to a debug log, if anything.

Why? The users should be told once, before the loop runs, that errors will be ignored. If someone had run this against 100 or so containers and the command failed (not in the `PATH` or such), then getting the warning that the errors were ignored 100 times is perhaps not so useful.

A small UI/UX change. Not quite the Unix philosophy, though.

Reply:
Sounds good, changed to become a debug message.

Reply:
What I meant would be something akin to this:

```console
$ cictl exec -p -x --label='test=123' date
WARN[0000] The `--ignore-errors` option has been set. Errors following command execution will be ignored...
...
...
...
```

Perhaps something with better wording.

Reply:
Got it, added that additional log line.